File: Python/Recursivos/BubbleSort.py
def BubbleSort(vetor, tamanho):
	trocas = 0
	for x in xrange(0, tamanho-1):
		if vetor[x] > vetor[x+1]:
			temp = vetor[x]
			vetor[x] = vetor[x+1]
			vetor[x+1] = temp
			trocas = 1
	if trocas != 0:
		BubbleSort(vetor, tamanho-1)
	return vetor

vetor = [2, 9, 8, 0, 1, 3, 5, 4, 6, 7]
print vetor
vetor = BubbleSort(vetor, len(vetor))
print vetor

File: src/go/bubbleSort/bubbleSort.go

/*
 * Bubble Sort é um dos algoritmos de ordenação mais simples.
 * A ideia principal é percorrer o vetor/slice várias vezes
 * empurrando/flutuando os valores mais altos para o fim do vetor.
 */
package bubblesort

// Bubble Sort tradicional
func BubbleSort(slice []int) {
	for indice1 := len(slice) - 1; indice1 > 0; indice1-- {
		for indice2 := 0; indice2 < indice1; indice2++ {
			if slice[indice2] > slice[indice2+1] {
				temp := slice[indice2]
				slice[indice2] = slice[indice2+1]
				slice[indice2+1] = temp
			}
		}
	}
}

// Bubble Sort recursivo
func BubbleSortRecursivo(slice []int, tamanho int) {
	trocas := 0
	for indice := 0; indice < tamanho-1; indice++ {
		if slice[indice] > slice[indice+1] {
			temp := slice[indice]
			slice[indice] = slice[indice+1]
			slice[indice+1] = temp
			trocas++
		}
	}
	if trocas != 0 {
		BubbleSortRecursivo(slice, tamanho-1)
	}
}
